### ISSUE TITLE
Google Ends Rich Results for Data-Vocabulary.org Structured Data - transition to Schema.org

### ISSUE THREAD

**Issue by ondrejvysek** [getgrav/grav-plugin-breadcrumbs#25]
Hi,
many of you probably received an infotmation from google search console about error regarding Data-Vocabulary.org is being deprecated. here is the quick fix (it is really quick, spent 20min with it, not tested widely) to this to switch from Data-Vocabulary.org to the Shema.org which is supportted by Google now. It consists of 2 modifications:
**1. breadcrumbs/templates/partials/breadcrumbs.html.twig**
```
{% set crumbs = breadcrumbs.get() %}
{% set breadcrumbs_config = config.plugins.breadcrumbs %}
{% set divider = breadcrumbs_config.icon_divider_classes %}
{% set lang = "/"~html_lang %}
{% set siteurl = base_url_absolute|replace({ (lang) : ""}) %}

{% if crumbs|length > 1 or breadcrumbs_config.show_all %}
<ol id="breadcrumbs" itemscope itemtype="https://schema.org/Breadcrumblist">
    {% if breadcrumbs_config.icon_home %}
    <i class="{{ breadcrumbs_config.icon_home }}"></i>
    {% endif %}
	{% set position = 1 %}
    {% for crumb in crumbs %}
        {% if not loop.last %}
            {% if crumb.routable %}
			<li itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem">
                <a href="{{ siteurl }}{{ crumb.url }}" itemprop="item"><span itemprop="name">{{ crumb.menu }}</span></a>
				<meta itemprop="position" content="{{ position }}">
			</li>
            {% else  %}
			<li itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem">
                <span itemprop="name">{{ crumb.menu }}</span>
				<meta itemprop="position" content="{{ position }}">
			</li>
            {% endif %}
            <i class="{{ divider }}"></i>
        {% else %}
            {% if breadcrumbs_config.link_trailing %}
			<li itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem">
                <a href="{{ siteurl }}{{ crumb.url }}" itemprop="item"><span itemprop="name">{{ crumb.menu }}</span></a>
				<meta itemprop="position" content="{{ position }}">
			</li>
            {% else %}
			<li itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem">
                <span itemprop="name">{{ crumb.menu }}</span>
				<meta itemprop="position" content="{{ position }}">
			</li>
            {% endif %}
        {% endif %}
		{% set position = position + 1 %}
    {% endfor %}
</ol>
{% endif %}
```
**2. breadcrumbs/css/breadcrumbs.css**
```
#breadcrumbs {
    height: 3rem;
    line-height: 3rem;
    padding-left: 3rem;
}
#breadcrumbs li {
    display: inline-block;
    padding: 0 1rem;
}
#breadcrumbs ol {
    padding-left: 0;list-style:none;display:-webkit-inline-box;
}

#breadcrumbs i {
    display: inline-block;
    width: 20px;
    text-align: center;
} 
```

this will mage Google happy and you site will look exactly the same. You may test the structured data result here https://search.google.com/structured-data/testing-tool

**Comment by ondrejvysek** [comment]
Small modification to include only routable pages in the metadata, non-routable are just displayed on the page but ignored in the metadata.
```
{% set crumbs = breadcrumbs.get() %}
{% set breadcrumbs_config = config.plugins.breadcrumbs %}
{% set divider = breadcrumbs_config.icon_divider_classes %}
{% set lang = "/"~html_lang %}
{% set siteurl = base_url_absolute|replace({ (lang) : ""}) %}

{% if crumbs|length > 1 or breadcrumbs_config.show_all %}
<ol id="breadcrumbs" itemscope itemtype="https://schema.org/Breadcrumblist">
    {% if breadcrumbs_config.icon_home %}
    <i class="{{ breadcrumbs_config.icon_home }}"></i>
    {% endif %}
	{% set position = 1 %}
    {% for crumb in crumbs %}
        {% if not loop.last %}
            {% if crumb.routable %}
			<li itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem">
                <a href="{{ siteurl }}{{ crumb.url }}" itemprop="item"><span itemprop="name">{{ crumb.menu }}</span></a>
				<meta itemprop="position" content="{{ position }}">
				{% set position = position + 1 %}
			</li>
            {% else  %}
			<li>
                <span>{{ crumb.menu }}</span>
				
			</li>
            {% endif %}
            <i class="{{ divider }}"></i>
        {% else %}
            {% if breadcrumbs_config.link_trailing %}
			<li itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem">
                <a href="{{ siteurl }}{{ crumb.url }}" itemprop="item"><span itemprop="name">{{ crumb.menu }}</span></a>
				<meta itemprop="position" content="{{ position }}">
				{% set position = position + 1 %}
			</li>
            {% else %}
			<li>
                <span>{{ crumb.menu }}</span>
				
			</li>
            {% endif %}
        {% endif %}
		
    {% endfor %}
</ol>
{% endif %}
```

**Comment by nex84** [comment]
#29

**Comment by rhukster** [comment]
Pull request please!

**Comment by rhukster** [comment]
Released as 1.6.0
